starter-code/HeterogeneousRecyclerView/app/src/main/java/com/charlesdrews/heterogeneousrecyclerview/FootballRecyclerViewAdapter.java

```diff
@@ -11,12 +11,54 @@
 
 import java.util.List;
 
+import javax.xml.transform.Templates;
+
 /**
  * Created by charlie on 11/14/16.
  */
 
 public class FootballRecyclerViewAdapter extends RecyclerView.Adapter {
+    private List<BaseFootballObject> mBaseFootballObjects;
+
+    public FootballRecyclerViewAdapter(List<BaseFootballObject> baseFootballObjects) {
+        mBaseFootballObjects = baseFootballObjects;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        if (viewType==1) {
+            return new PlayerViewHolder(
+                    LayoutInflater.from(parent.getContext())
+                            .inflate(R.layout.player_list_entry, parent, false));
+        }
+
+        return new TeamViewHolder(
+                LayoutInflater.from(parent.getContext())
+                        .inflate(R.layout.team_list_entry,parent,false)
+        );
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        if (holder instanceof PlayerViewHolder){
```

Comment:
This works, but it might be slightly faster to call `holder.getItemViewType()` to get the `int` representing the view type. The `instanceof` operator relies on _reflection_ which is slow.